bedrock/entity/workflows.py

```diff
@@ -0,0 +1,169 @@
+import asyncio
+from collections import deque
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Deque, List, Optional, Tuple
+
+from temporalio import workflow
+
+with workflow.unsafe.imports_passed_through():
+    from bedrock.shared.activities import BedrockActivities
+
+
+@dataclass
+class BedrockParams:
+    conversation_summary: Optional[str] = None
+    prompt_queue: Optional[Deque[str]] = None
+
+
+@workflow.defn
+class EntityBedrockWorkflow:
+    def __init__(self) -> None:
+        # List to store prompt history
+        self.conversation_history: List[Tuple[str, str]] = []
+        self.prompt_queue: Deque[str] = deque()
+        self.conversation_summary: Optional[str] = None
+        self.continue_as_new_per_turns: int = 6
+        self.chat_ended: bool = False
+
+    @workflow.run
+    async def run(
+        self,
+        params: BedrockParams,
+    ) -> str:
+
+        if params and params.conversation_summary:
+            self.conversation_history.append(
+                ("conversation_summary", params.conversation_summary)
+            )
+
+            self.conversation_summary = params.conversation_summary
+
+        if params and params.prompt_queue:
+            self.prompt_queue.extend(params.prompt_queue)
+
+        while True:
+            workflow.logger.info("Waiting for prompts...")
+
+            # Wait for a chat message (signal) or timeout
+            await workflow.wait_condition(
+                lambda: bool(self.prompt_queue) or self.chat_ended
+            )
+
+            if bool(self.prompt_queue):
```

Comment:
```suggestion
            if self.prompt_queue:
```

Shouldn't need to convert to `bool` here